go/atomic2/atomic128_test.go

```diff
@@ -24,6 +24,17 @@ import (
 )
 
 func TestCompareAndSwap(t *testing.T) {
+	runCompareAndSwap(t)
+}
+
+func TestCompareAndSwapWriteBarrier(t *testing.T) {
+	writeBarrier.enabled = true
```

Comment:
This modifies global state and at least we need to revert it back at the end of the test, but it also could introduce risk for flaky tests if anything runs in parallel. 

Also this links to runtime stuff, so I think we're tbh probably better off not adding this test here. Because this hooks into the Go runtime.

Reply:
I understand, this test was made just to cover [this conditional](https://github.com/vitessio/vitess/blob/1b328bffb853ed08da621a7d144a01e06a6cf8d3/go/atomic2/atomic128.go#L55C5-L55C5). Is there a way to test it without modifying global state?

Reply:
I don’t think there’s a good way to test so I think we should skip trying to further test this. 

Code coverage is a tool to find areas that might need more tests. It’s not a goal in itself to get too 100% coverage. That can have detrimental effects and this is one of those cases.

Reply:
Okay, I'll close this PR